stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/ChannelControllerImplEventNewTest.kt

```diff
@@ -0,0 +1,144 @@
+package io.getstream.chat.android.livedata.controller
+
+import androidx.lifecycle.Observer
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.never
+import com.nhaarman.mockitokotlin2.verify
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.events.NewMessageEvent
+import io.getstream.chat.android.client.events.UserStartWatchingEvent
+import io.getstream.chat.android.client.models.Config
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.livedata.ChatDomainImpl
+import io.getstream.chat.android.test.InstantTaskExecutorExtension
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import java.util.Date
+
+private const val CURRENT_USER_ID = "currentUserId"
+
+@ExperimentalCoroutinesApi
+@ExtendWith(InstantTaskExecutorExtension::class)
+internal class ChannelControllerImplEventNewTest {
+
+    private val chatClient: ChatClient = mock()
+    private val chatDomain: ChatDomainImpl = mock {
+        on(it.scope) doReturn TestCoroutineScope()
+        on(it.currentUser) doReturn User(id = CURRENT_USER_ID)
+        on(it.getChannelConfig(any())) doReturn Config(isConnectEvents = true, isMutes = true)
+    }
+
+    private lateinit var channelControllerImpl: ChannelControllerImpl
+
+    @BeforeEach
+    fun setUp() {
+        channelControllerImpl = ChannelControllerImpl(
+            channelType = "type1",
+            channelId = "channelId",
+            client = chatClient,
+            domainImpl = chatDomain
+        )
+    }
+
+    @Test
+    fun `when user watching event arrives, watchers should be incremented`() = runBlockingTest {
+        val user = User()
+
+        val userStartWatchingEvent = UserStartWatchingEvent(
```

Comment:
Could we have "Mother Methods" for that?

Reply:
What do you mean by "Mother Methods"?